projects/com.oracle.truffle.llvm.asm.amd64/src/com/oracle/truffle/llvm/asm/amd64/AsmFactory.java

```diff
@@ -673,10 +985,127 @@ private static PrimitiveType getPrimitiveTypeFromSuffix(char suffix) {
         }
     }
 
+    private class XchgOperands {
```

Comment:
As this file becomes quite long, you might consider moving this code to a package protected class in a separate file?